### PR TITLE
quant: rename _ObserverBase to UniformQuantizationObserverBase

### PR DESCRIPTION
**Pull request by vkuzo**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#76461 quant: rename _ObserverBase to UniformQuantizationObserverBase**
* #76460 quant: make RecordingObserver inherit from ObserverBase

Summary:

Renaming as the old name was confusing. The name represents
better what this class is doing.

Test plan:

CI

Differential Revision: [D35976350](https://our.internmc.facebook.com/intern/diff/D35976350)

### CODE REVIEW

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35976350).

File: torch/ao/quantization/observer.py [hunk 1]
```diff
@@ -173,7 +174,7 @@ def __init__(
         eps=torch.finfo(torch.float32).eps,
     ) -> None:
         factory_kwargs = torch.nn.factory_kwargs(factory_kwargs)
-        super(_ObserverBase, self).__init__(dtype=dtype)
+        super(UniformQuantizationObserverBase, self).__init__(dtype=dtype)
```
Comment (jerryzh168) [inline torch/ao/quantization/observer.py:177]: this can be `super().__init__(dtype=dtype)` now I think

File: torch/ao/quantization/observer.py [hunk 2]
```diff
@@ -334,7 +335,12 @@ def reset_min_max_vals(self):
         raise NotImplementedError("Cannot reset min/max values in the given observer.")
 
 
-class MinMaxObserver(_ObserverBase):
+# Originally, this class was called `_ObserverBase`.  Keeping the old name around
+# for backwards compatibility.
```
Comment (jerryzh168) [inline torch/ao/quantization/observer.py:339]: should we add a TODO to remove it after certain release like 1.13?

**Review by jerryzh168 — APPROVED** [review]

File: torch/ao/quantization/observer.py (hunk 2 above)
Comment (vkuzo) [reply torch/ao/quantization/observer.py:339]: sure

File: torch/ao/quantization/observer.py (hunk 1 above)
Comment (vkuzo) [reply torch/ao/quantization/observer.py:177]: that's true, I just did a string replace without trying to change anything else (to minimize time + risk), let me know if you think this additional change is worth adding to this PR - I'm flexible

Comment (jerryzh168) [reply torch/ao/quantization/observer.py]: sounds good, I don't have a strong preference, we can do this later

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35976350).

**Pull request closed without merging**